sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -115,13 +116,15 @@
    */
   private final MemoryMode MEMORY_MODE;
 
-  public VectorizedParquetRecordReader(TimeZone convertTz, boolean useOffHeap) {
+  public VectorizedParquetRecordReader(TimeZone convertTz, boolean useOffHeap, int capacity) {
     this.convertTz = convertTz;
     MEMORY_MODE = useOffHeap ? MemoryMode.OFF_HEAP : MemoryMode.ON_HEAP;
+    this.capacity = capacity;
   }
 
+  // Vectorized parquet reader used for testing and benchmark.
   public VectorizedParquetRecordReader(boolean useOffHeap) {
-    this(null, useOffHeap);
+    this(null, useOffHeap, 4096);
```

Comment:
How about changing benchmark and test programs to pass capacity and remove this constructor?
These programs also have `SQLConf`.